src/datasources/asset/AssetDataSource.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  DataFrameDTO,
+  DataQueryRequest,
+  DataSourceInstanceSettings,
+  TestDataSourceResponse,
+} from '@grafana/data';
+import { BackendSrv, getBackendSrv, getTemplateSrv, TemplateSrv } from '@grafana/runtime';
+import { DataSourceBase } from 'core/DataSourceBase';
+import {
+  AssetFilterProperties,
+  AssetModel, AssetQuery,
+  AssetQueryType,
+  AssetsResponse,
+} from './types';
+import { getWorkspaceName, replaceVariables } from "../../core/utils";
+import { SystemMetadata } from "../system/types";
+import { defaultOrderBy, defaultProjection } from "../system/constants";
+
+export class AssetDataSource extends DataSourceBase<AssetQuery> {
+  constructor(
+    readonly instanceSettings: DataSourceInstanceSettings,
+    readonly backendSrv: BackendSrv = getBackendSrv(),
+    readonly templateSrv: TemplateSrv = getTemplateSrv()
+  ) {
+    super(instanceSettings, backendSrv, templateSrv);
+  }
+
+  baseUrl = this.instanceSettings.url + '/niapm/v1';
+
+  defaultQuery = {
+    type: AssetQueryType.Metadata,
+    workspace: '',
+    minionIds: [],
+  };
+
+  async runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+    return await this.processMetadataQuery(query)
+  }
+
+  async processMetadataQuery(query: AssetQuery) {
+    const result: DataFrameDTO = { refId: query.refId, fields: [] };
+    let workspaceId = this.templateSrv.replace(query.workspace);
+    const minionIds = replaceVariables(query.minionIds, this.templateSrv);
+    const conditions = [];
+    if (workspaceId) {
```

Comment:
Minor: you could move workspaceId query after the minionIds, since workspace is not indexed